packages/video_player_videohole/tizen/src/drm_manager.cc

```diff
@@ -0,0 +1,255 @@
+#include "log.h"
+
+/**
+    DRM Type is DM_TYPE_EME,
+    "com.microsoft.playready" => PLAYREADY
+    "com.widevine.alpha" => Wideveine CDM
+    "org.w3.clearkey" => Clear Key
+    "org.w3.cdrmv1"  => ChinaDRM
+*/
+static std::string GetDrmSubType(int dm_type) {
+  switch (dm_type) {
+    case 1:
+      return "com.microsoft.playready";
+    case 2:
+      return "com.widevine.alpha";
+    default:
+      return "com.widevine.alpha";
+  }
+}
+
+DrmManager::DrmManager(int drmType, const std::string &licenseUrl,
+                       player_h player)
+    : drm_type_(drmType), license_url_(licenseUrl), player_(player) {}
+
+DrmManager::~DrmManager() {}
+
+bool DrmManager::InitializeDrmSession(const std::string &url) {
+  int ret = 0;
+  drm_manager_handle_ = OpenDrmManager();
+  if (drm_manager_handle_ == nullptr) {
+    LOG_ERROR("Fail to open drm manager");
+    return false;
+  }
+
+  media_player_handle_ = OpenMediaPlayer();
+  if (media_player_handle_ == nullptr) {
+    LOG_ERROR("Fail to open libcapi-media-player.so.");
+    return false;
+  }
+
+  if (!CreateDrmSession()) {
+    LOG_ERROR("Fail to create drm session");
+    return false;
+  }
+
+  if (!SetPlayerDrm(url)) {
+    LOG_ERROR("Fail to set player drm");
+    return false;
+  }
+
+  if (!SetChallengeCondition()) {
+    LOG_ERROR("Fail to set challenge condition");
+    return false;
+  }
+
+  ret = DMGRSetData(drm_manager_handle_, drm_session_, "Initialize", nullptr);
+  if (ret != DM_ERROR_NONE) {
+    LOG_ERROR("initialize failed");
+    CloseDrmManager(drm_manager_handle_);
+    drm_manager_handle_ = nullptr;
+    drm_session_ = nullptr;
+    return false;
+  }
+  return true;
+}
+
+bool DrmManager::CreateDrmSession(void) {
+  LOG_INFO("dm_str: %s", GetDrmSubType(drm_type_).c_str());
+  drm_session_ = DMGRCreateDRMSession(drm_manager_handle_, DM_TYPE_EME,
+                                      GetDrmSubType(drm_type_).c_str());
+  LOG_INFO("drm_session_ id: %p", drm_session_);
+  return true;
+}
+
+bool DrmManager::SetPlayerDrm(const std::string &url) {
+  int ret = 0;
+  int drm_handle = 0;
+  ret = DMGRSetData(drm_manager_handle_, drm_session_, "set_playready_manifest",
+                    (void *)url.c_str());
+
+  SetDataParam_t configure_param;
+  configure_param.param1 = (void *)OnDrmManagerError;
+  configure_param.param2 = (void *)drm_session_;
+  ret = DMGRSetData(drm_manager_handle_, drm_session_,
+                    (char *)"error_event_callback", (void *)&configure_param);
+  if (ret != DM_ERROR_NONE) {
+    LOG_ERROR("setdata failed for renew callback:%s", get_error_message(ret));
+    return false;
+  }
+
+  ret = DMGRGetData(drm_manager_handle_, drm_session_, "drm_handle",
+                    (void **)&drm_handle);
+  if (ret != DM_ERROR_NONE) {
+    LOG_ERROR("DMGRGetData drm_handle failed:%s", get_error_message(ret));
+    return false;
+  }
+  LOG_DEBUG("DMGRGetData drm_handle succeed, drm handle: %d", drm_handle);
+
+  if (player_set_drm_handle) {
```

Comment:
This condition always evaluate to 'true'

Reply:
> This condition always evaluate to 'true'

have fixed